src/detail/fastrtps_dynamic_data.cpp

```diff
@@ -444,7 +449,7 @@ fastrtps__dynamic_data_get_string_value(
   );
 
   *value_length = tmp_string.size();
-  char * tmp_out = new char[*value_length + 1];
+  char * tmp_out = new char[*value_length + 1];  // TODO(methylDragon): Use alloc here
```

Comment:
I'm not sure which `alloc` you are referring to here.  Can you be more specific?  I would think we would need to pass in an allocator and use that, but I'm not sure.

Further, we need to check for allocation failure here.  Either we need to specify `nothrow` and check for NULL, or capture the exception (which won't be able to bubble through the C layer that calls into this).

Same goes a few times below.

Reply:
Just the `new`, yeah.. Because the caller will be in charge of destroying the obtained buffer

I was thinking it'd be good if they did it using the allocator stored in the data_impl, but because it was an implementation detail, I didn't want to get to it yet

Reply:
This one I will defer for now (hence the TODO)